pyxis_slurmstepd.c

```diff
@@ -251,6 +264,85 @@ static int enroot_create_user_runtime_dir(void)
 	return (0);
 }
 
+/* As root, create the per-user directory where squashfs files are stored and later could be reused. */
+static int enroot_create_image_save_dir(char *path) 
+{
+	int ret;
+	char uid_dir_path[PATH_MAX];
+
+	if (path == NULL || strlen(path) == 0) {
+		slurm_error("pyxis: path is empty");
+		return (-1);
+	}
+
+	/* In case it's a dir, we create per user folder */
+	if (path[strlen(path) - 1] == '/') {
+		ret = snprintf(uid_dir_path, sizeof(uid_dir_path), "%s%u/", path, context.job.uid);
+		if (ret < 0 || ret >= sizeof(uid_dir_path)) {
+			slurm_error("pyxis: too long path to save image: %s", path);
+			return (-1);
+		}
+		path = uid_dir_path;
+	}
+
+	/* Start from 1 to avoid first / */
+	for (size_t i = 1; i < strlen(path); i++) {
```

Comment:
are spaces possible at the beginning? if so, i'd skip them and '/' in a separate loop

Reply:
added validation in `srun`